sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -48,6 +48,15 @@ public SentryId captureEvent(SentryEvent event, @Nullable Scope scope) {
       event.setEnvironment(options.getEnvironment());
     }
 
+    if (scope != null) {
```

Comment:
* scope level has priority over event level
* move block to MainProcessor
* List should be null by default and check that before setting values